backend/kernelCI_app/cache.py

```diff
@@ -0,0 +1,83 @@
+from django.core.cache import cache
+from django.db import connection
+from threading import Thread
+
+
+DEFAULT_QUERY_CACHE = 180
+NOTIFY_CHANNEL = "teste"
+
+__commit_lookup = {}
+__build_lookup = {}
+__test_lookup = {}
+
+
+def __createCacheParamsHash(params: dict):
+    params_list = list(params.items())
+    params_list.sort(key=lambda x: x[0])
+    params_string = ",".join([i[1] for i in params_list])
+    return hash(params_string)
+
+
+def setQueryCache(key, params, rows, commit_hash=None, build_id=None, test_id=None,
+                  timeout=DEFAULT_QUERY_CACHE):
+    params_hash = __createCacheParamsHash(params)
+    hash_key = "%s-%s" % (key, hash(params_hash))
+
+    __addToLookup(hash_key, commit_hash, __commit_lookup)
+    __addToLookup(hash_key, build_id, __build_lookup)
+    __addToLookup(hash_key, test_id, __test_lookup)
```

Comment:
this seems like a modular utility for caching function but this section is making it specific to the tests/boots, at least add a todo here